.github/CONTRIBUTING.md

```diff
@@ -1,6 +1,130 @@
+
+**References:** If you are not sure how to issue a PR, but would like to contribute, these links should help get you started:
+
+1. **[How to Github: Fork, Branch, Track, Squash and Pull request](https://gun.io/blog/how-to-github-fork-branch-and-pull-request/)**.
+2. **[Squashing Github pull requests into a single commit](http://eli.thegreenplace.net/2014/02/19/squashing-github-pull-requests-into-a-single-commit)**.
+3. **[Github help](https://help.github.com/articles/using-pull-requests/)** - you'll need the *fork and pull* model.
+
+Minimal first time PR
+---------------------
+
+```
+$ cd /tmp      # or anywhere safe to play
+$ git config --global core.autocrlf false   # Windows-only preserve \n in test data
+$ git clone https://github.com/Rdatatable/data.table.git
+$ cd data.table
+$ R CMD build .
+$ R CMD check data.table_1.11.5.tar.gz
+...
+Status: OK
+```
+Congratulations - you've just compiled and tested the very latest version of data.table in development. Everything looks good. Now make your changes. Using an editor of your choice, edit the appropriate `.R`, `.md`, `NEWS` and `tests.Rraw` files. Test your changes : 
+```
+$ R CMD build .
+$ R CMD check data.table_1.11.5.tar.gz
+```
+or if your OS supports makefile
+```
+$ make build && make check
+```
+Fix the problems and repeat the `build` and `check` steps until you get `Status: OK`.
+Now commit the change and push. Since this is a first time PR and you're not a project member, this step should automatically ask you if you wish to fork the project. Say 'yes'. If that's not the case, please edit this wiki page to show what exactly happens for non project members.
+```
+$ git commit -am "Added/fixed something in somewhere"
+$ git push
+```
+After your first successful non-trivial PR you'll likely be invited to be a project member. When you're a member, before making your changes, you would create a branch first : 
+```
+$ git checkout -b my_new_branch
+```
+and then the `commit` and `push` shown above would push to the branch in the main project. The next time you refresh the GitHub page in your browser, a button appears which you can click to create the PR from the branch. And that's all there is to it.
+
+#### Translations
+
+`data.table` offers some translations so that our users can get feedback (errors, warnings, verbose messages) in their native language. Currently we only support Mandarin Chinese.
+
+The data for these translations lives in the `po` folder. You do not need to make any changes here for your PR -- translations are updated in a batch before each CRAN release.
+
+If you are writing R code, please avoid breaking up strings into multiple parts (i.e., wide lines are OK for error messages).
+
+If you are writing C code, please wrap your message strings (`char` arrays) with `_()`, e.g. `error(_("step %d failed"), ii)` instead of `error("step %d failed", ii)`. Common functions for this are `error`, `warning`, and `Rprintf`.
```

Comment:
this "translations" sections sounds more friendly than the previous "translations" section. and of course we don't need two.

Reply:
I think I forgot to delete the old section when I added a re-write to the Wiki. Removed the first one.